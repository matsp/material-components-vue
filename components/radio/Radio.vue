<template>
  <div
    class="mdc-radio"
    :class="classes"
  >
    <input
      ref="input"
      class="mdc-radio__native-control"
      type="radio"
      v-bind="attrs"
      v-on="listeners"
      @change="$emit('change', $event.target.value)"
    >
    <div class="mdc-radio__background">
      <div class="mdc-radio__outer-circle" />
      <div class="mdc-radio__inner-circle" />
    </div>
  </div>
</template>

<script>
import { MDCRadio } from '@material/radio'

import { baseComponentMixin, themeClassMixin } from '@components/base'

export default {
  mixins: [baseComponentMixin, themeClassMixin],
  inject: {
    formFieldInputAssigning: {
      default: null
    }
  },
  model: {
    prop: 'picked',
    event: 'change'
  },
  props: {
    js: {
      type: Boolean,
      default: true
    }
  },
  data () {
    return {
      mdcRadio: undefined
    }
  },
  computed: {
    classes () {
      return {
        'mdc-radio--disabled': this.$attrs.disabled != null
      }
    },
    attrs () {
      const _attrs = this.$attrs
      delete _attrs.picked
      return _attrs
    },
    listeners () {
      const _listeners = this.$listeners
      delete _listeners.change
      return _listeners
    }
  },
  watch: {
    js () {
      this.reInstantiate()
    }
  },
  mounted () {
    if (this.js) {
      this.mdcRadio = MDCRadio.attachTo(this.$el)
      if (this.formFieldInputAssigning instanceof Function) {
        this.formFieldInputAssigning(this.mdcRadio)
      }
    }
  },
  beforeDestroy () {
    if (this.js) {
      this.mdcRadio.destroy()
      if (this.formFieldInputAssigning instanceof Function) {
        this.formFieldInputAssigning(undefined)
      }
    }
  },
  methods: {
    reInstantiate () {
      if (this.mdcRadio instanceof MDCRadio) {
        this.mdcRadio.destroy()
        this.mdcRadio = undefined
      }
      if (this.js) {
        MDCRadio.attachTo(this.$el)
        if (this.formFieldInputAssigning instanceof Function) {
          this.formFieldInputAssigning(this.mdcRadio)
        }
      } else {
        this.mdcRadio = undefined
      }
    }
  }
}
</script>
