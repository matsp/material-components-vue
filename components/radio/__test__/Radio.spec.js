import 'mutationobserver-shim'
import { mount } from '@vue/test-utils'
import Radio from '../Radio.vue'

describe('Checkbox', () => {
  it('should mount', () => {
    let wrapper = mount(Radio)
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.vm.$data.mdcRadio).toBeDefined()
  })

  it('should render with no prop', () => {
    let wrapper = mount(Radio)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-radio')
    expect(wrapper.find('input').attributes('disabled')).toBeUndefined()
  })

  it('should render as disabled', () => {
    let wrapper = mount(Radio, {
      propsData: {
        disabled: true
      }
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.classes()).toContain('mdc-radio--disabled')
    expect(wrapper.find('input').attributes('disabled')).toBeDefined()
  })

  it('should render and emit', () => {
    let wrapper = mount(Radio, {
      propsData: {
        value: 'val'
      }
    })

    const radio = wrapper.find('input')
    radio.setChecked(true)
    expect(wrapper.emitted().change.length).toBe(1)
    expect(wrapper.emitted().change[0]).toEqual(['val'])
  })
})
