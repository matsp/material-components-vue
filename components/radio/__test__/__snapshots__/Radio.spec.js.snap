// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render as disabled 1`] = `
<div class="mdc-radio mdc-radio--disabled"><input type="radio" class="mdc-radio__native-control" disabled="" value="">
  <div class="mdc-radio__background">
    <div class="mdc-radio__outer-circle"></div>
    <div class="mdc-radio__inner-circle"></div>
  </div>
</div>
`;

exports[`Checkbox should render with no prop 1`] = `
<div class="mdc-radio"><input type="radio" class="mdc-radio__native-control" value="">
  <div class="mdc-radio__background">
    <div class="mdc-radio__outer-circle"></div>
    <div class="mdc-radio__inner-circle"></div>
  </div>
</div>
`;
